### PR TITLE
Clean up example samples

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This PR cleans up the "complete" and "autoscaler" sample yamls a bit.
Unnecessary pod spec fields are removed without sacrificing the completeness of the examples. 
The idea is to make the configuration look less intimidating.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/config/samples/ray-cluster.autoscaler.large.yaml [hunk 1]
```diff
@@ -61,61 +61,22 @@ spec:
...
     rayStartParams:
-      # Flag "no-monitor" must be set when running the autoscaler in
-      # a sidecar container.
-      port: '6379'
+      # Flag "no-monitor" will be automatically set when autoscaling is enabled.
       dashboard-host: '0.0.0.0'
-      node-ip-address: $MY_POD_IP # auto-completed as the head pod IP
       block: 'true'
-      num-cpus: '1' # can be auto-completed from the limits
+      # num-cpus: '1' # can be auto-completed from the limits
```
Comment (DmitriGekhtman) [inline ray-operator/config/samples/ray-cluster.autoscaler.large.yaml:67]: This is a foot-gun for people copying the config and modifying resource values.

**Review by Jeffwan — APPROVED** [review]

**Comment by Jeffwan** [comment]
great. These configuration are covered by codes. Looks good to me

**Comment by DmitriGekhtman** [comment]
We could consider automatically configuring the init container and pre-stop hook in the 0.4.0 release.

**Review by brucez-anyscale — APPROVED** [review]
lgtm

**Comment by DmitriGekhtman** [comment]
Just going to double check that the configs don't have any issues before merging.

File: ray-operator/config/samples/ray-cluster.complete.yaml [hunk 2]
```diff
@@ -44,32 +38,7 @@ spec:
...
-                resource: limits.memory
-          - name: MEMORY_REQUESTS
-            valueFrom:
-              resourceFieldRef:
-                containerName: ray-head
-                resource: requests.memory
-          - name: MY_POD_IP
-            valueFrom:
-              fieldRef:
-                fieldPath: status.podIP
```
Comment (davidxia) [inline ray-operator/config/samples/ray-cluster.complete.yaml:72]: @DmitriGekhtman just curious if these env vars are deprecated and should no longer be used or if removing them will change Ray behavior? I'm using Ray 1.13.

File: ray-operator/config/samples/ray-cluster.complete.yaml [hunk 3]
```diff
@@ -116,56 +83,12 @@ spec:
...
-          - name: MY_POD_NAME
-            valueFrom:
-              fieldRef:
-                fieldPath: metadata.name
-          - name: MY_POD_IP
-            valueFrom:
-              fieldRef:
-                fieldPath: status.podIP
-          ports:
-          - containerPort: 80
```
Comment (davidxia) [inline ray-operator/config/samples/ray-cluster.complete.yaml:168]: @DmitriGekhtman what did this port used to be? Should we add the `metrics` port 8080?

File: ray-operator/config/samples/ray-cluster.complete.yaml (hunk 2 above)
Comment (DmitriGekhtman) [reply ray-operator/config/samples/ray-cluster.complete.yaml:72]: They're deprecated and can be safely removed.

File: ray-operator/config/samples/ray-cluster.complete.yaml (hunk 3 above)
Comment (DmitriGekhtman) [reply ray-operator/config/samples/ray-cluster.complete.yaml:168]: I have no idea what this port was doing here.

Comment (DmitriGekhtman) [reply ray-operator/config/samples/ray-cluster.complete.yaml:168]: I don't believe that the operator interacts with ports specified in workerGroupSpecs.

**Pull request merged**